netcore/System.Private.CoreLib/src/System.Reflection.Emit/DynamicMethod.cs

```diff
@@ -316,14 +317,25 @@ internal override int GetParametersCount ()
 		}
 
 		public override string ToString () {
-			string parms = String.Empty;
-			ParameterInfo[] p = GetParametersInternal ();
-			for (int i = 0; i < p.Length; ++i) {
+			var paramsSb = new StringBuilder ();
+			ParameterInfo [] parameters = GetParametersInternal ();
+			for (int i = 0; i < parameters.Length; ++i) {
+				ParameterInfo p = parameters [i];
 				if (i > 0)
-					parms = parms + ", ";
-				parms = parms + p [i].ParameterType.Name;
+					paramsSb.Append (", ");
+
+				paramsSb.Append (p.ParameterType.IsPrimitive ? p.ParameterType.Name : p.ParameterType.ToString ());
 			}
-			return ReturnType.Name+" "+Name+"("+parms+")";
+
+			string returnTypeName;
+			if (ReturnType == typeof (void))
+				returnTypeName = "Void";
+			else if (ReturnType.IsPrimitive)
+				returnTypeName = ReturnType.Name;
+			else
+				returnTypeName = ReturnType.ToString ();
```

Comment:
CoreCLR also has exception for `TypedReference` (https://github.com/dotnet/coreclr/blob/2ec0bcfefe0bccb9a7023b8d528c775349d93458/src/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs#L3904). Do we want to special case that one too?

Reply:
Good catch!